### PR TITLE
Fix LinkedIn connection bugs and surface errors

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

This surfaces errors during the Connect w/ LinkedIn flow to the user via a toast vs. not showing them anything. It also fixes a bug where the `<LinkedInCallback/>` should be moved up to the top of the render tree so it's not prevented by any other blocking flows.

## Related Issue(s)

Fixes WEB-341

## How to test

This one is tricky to test as it depends on an environment setup w/ the right secrets setup. Currently the preview environment for this PR has them setup, so you should be able to onboard and try to connect with linked in. Verify the flow works.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-web-3c8ac4154b2</li>
	<li><b>🔗 URL</b> - <a href="https://brad-web-3c8ac4154b2.preview.gitpod-dev.com/workspaces" target="_blank">brad-web-3c8ac4154b2.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - brad-web-341-update-deprecated-linkedin-api-methods-gha.12349</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/App.tsx
Comment (selfcontained) [inline components/dashboard/src/App.tsx:46]: Moving this to the top of the render tree so it's not prevented by any blocking flows.

**Review by easyCZ — APPROVED** [review]

**Comment by filiptronicek** [comment]
/hold 

to unblock merge

**Pull request merged**
